clo-workflows/Scripts/src/store/ClientStore.ts

```diff
@@ -180,14 +137,140 @@ export class ClientStore {
+            projectDetails.submitterId = this.currentUser.Id
+            projectDetails.type = this.viewState.selectedProjectType
+            await this.dataService.createProject(projectDetails)
+            runInAction(() => this.projects.push(projectDetails))
+            this.postMessage({ messageText: "project successfully created", messageType: MessageBarType.success })
+        } catch (error) {
+            console.log(error)
+            this.postMessage({ messageText: "there was a problem creating your new Project, try again", messageType: MessageBarType.error })
+        } finally {
+            this.setAsyncPendingLockout(false)
+            this.closeProjectModal()
+        }
+    }
+
+    @action
+    async submitNewWork(workDetails): Promise<void> {
+        this.setAsyncPendingLockout(true)
+        try {
+            workDetails.submitterId = this.currentUser.Id
+            workDetails.type = this.viewState.selectedWorkType
+            await this.dataService.createWork(workDetails)
+            this.postMessage({ messageText: "new work request successfully created", messageType: MessageBarType.success })
+        } catch (error) {
+            console.log(error)
+            this.postMessage({
+                messageText: "there was a problem submitting your new Work request, try again",
+                messageType: MessageBarType.error,
+            })
+        } finally {
+            this.setAsyncPendingLockout(false)
+            this.closeWorkModal()
+        }
+    }
+
+    @action
+    async submitProcess(processDetails): Promise<void> {
+        processDetails.submitterId = this.currentUser.Id
+        processDetails.type = this.viewState.selectedProjectType
+        await this.dataService.createProcess(processDetails)
+        this.closeProcessModal()
+    }
+    @action
+    closeProjectModal() {
+        this.newProject = observable.map()
+        this.selectedProjectType = undefined
+        this.showProjectModal = false
+    }
+
+    @action
+    closeProcessModal() {
+        this.newProcess = observable.map()
+        this.selectedWorkType = undefined
+        this.showProcessModal = false
+    }
+
+    @action
+    closeWorkModal() {
+        this.newWork = observable.map()
+        this.selectedWorkType = undefined
+        this.showWorkModal = false
+    }
+
+    @action
+    async fetchClientProjects() {
+        this.projects = await this.dataService.fetchClientProjects().then(projs => {
+            return projs.filter(p => {
+                return p.submitterId === this.currentUser.Id
+            })
+        })
+    }
+
+    @action
+    async fetchClientProcesses() {
+        this.processes = await this.dataService.fetchClientProcesses().then(projs => {
+            return projs.filter(p => {
+                return p.submitterId === this.currentUser.Id
+            })
+        })
+    }
+    @action
+    updateMember(m: string, v?: any) {
+        !v ? (this[m] = !this[m]) : (this[m] = v)
```

Comment:
this is not very safe or descriptive. This should probably be split into explicit actions so that its easier to see whats going on. Also avoid single letter parameter names